rust/src/mime/detect.rs

```diff
@@ -27,8 +27,16 @@ pub unsafe extern "C" fn SCDetectMimeEmailGetData(
     ctx: &MimeStateSMTP, buffer: *mut *const u8, buffer_len: *mut u32,
     hname: *const std::os::raw::c_char,
 ) -> u8 {
+    *buffer = ptr::null();
+    *buffer_len = 0;
+
     let c_str = CStr::from_ptr(hname); //unsafe
-    let str = c_str.to_str().unwrap_or("");
+    let str = match c_str.to_str() {
+        Ok(s) => s,
+        Err(_) => {
+            return 0;
```

Comment:
could also be a debug validation that it does not happen as we call `SCDetectMimeEmailGetData` with ascii strings

Reply:
You mean like this? @catenacyber 
```
let Ok(str) = c_str.to_str() else {
        SCLogDebug!("Received non-UTF8 string in SCDetectMimeEmailGetData");
        return 0;
    };
```